engineering-decision-records/010-ledger-validation-context.md

```diff
@@ -0,0 +1,321 @@
+---
+type: architecture
+status: accepted
+---
+
+# Ledger validation context
+
+## Context
+
+The Cardano ledger is a Mealy machine whose transitions are formally specified. In Hindley-Milner notation, each transition roughly resembles:
+
+```
+st -> cmd -> (st, events)
+```
+
+Where:
+
+- `st` designates the entire ledger state or a subset.
+- `cmd` designates a block, a transaction or part of a transaction.
+
+Cardano transactions are, in fact, akin to a collection of primitive operations that affect the state and a certain number of pre- and post-conditions. They are structured in fields and have a rather precise validation sequence order.
+
+The entire validation sequence can be [mapped as a flow chart](https://app.excalidraw.com/l/7Ao24g28S3o/872UZdalVsy), which reads top-down and left-to-right.
```

Comment:
We should link to [here](https://cardano-scaling.github.io/cardano-blueprint/ledger/block-validation.html) instead of the Excalidraw, it's more permanent (and has other benefits that help the blueprint)

Reply:
But the diagram on the blueprint is objectively worse 😅... it's harder to navigate and more importantly, it's missing all the links to the source code which add a lot of value.

Reply:
I strongly agree with you 😛  
Mermaid + mdbook vs Excalidraw I guess?


My main concern with the Excalidraw is that it belongs to the sundae org and is inherently less "long-lived" than the blueprint. Also worth noting that the Excalidraw link gives the visitor edit permissions, which can cause problems...

I agree the links are vital. But aren't you suggesting improvements to the blueprint that should be added instead of reasons the Excalidraw is better? 😉 

I guess we should defer to @Quantumplation and @ch1bo on that